common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -174,11 +174,6 @@ public static float getFloat(Object object, long offset) {
   }
 
   public static void putFloat(Object object, long offset, float value) {
-    if (Float.isNaN(value)) {
-      value = Float.NaN;
-    } else if (value == -0.0f) {
-      value = 0.0f;
-    }
```

Comment:
These change are expected to cause the following test case failure in `PlatformUtilSuite`, but it seems to be missed. Could you fix the test case or remove together, @cloud-fan ?
- https://github.com/apache/spark/blob/master/common/unsafe/src/test/java/org/apache/spark/unsafe/PlatformUtilSuite.java#L162-L163